src/main/scala/com/fulcrumgenomics/commons/io/Io.scala

```diff
@@ -226,4 +226,10 @@ trait IoUtil {
     stream.safelyClose()
     array
   }
+
+  /** An iterable view of source data.  The transformations are non-strict, and the source is able to be closed.  */
+  trait LazySource[A] extends View[A] with Closeable {
```

Comment:
go crazy with the name, it just can't be `Source`

Reply:
Hehehe.  I think the name is fine.  I think I would just extract it to a top level class and put it alongside `Writer`.